ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/KVUtilsTransactionSpec.scala

```diff
@@ -106,6 +106,17 @@ class KVUtilsTransactionSpec extends AnyWordSpec with Matchers with Inside {
         }
     }
 
+    "be able to pre-execute an exercise by key" in
```

Comment:
I would change description of this test case to be less generic and reflect what we discussed in the PR thread, e.g. `be able to pre-execute an exercise by key in a transaction with key replacement`.

Reply:
What do you mean by "with key replacement"? The key gets updated whenever the contract it points to gets updated, so if you mean that, it is implicit in the definition of exercising a (consuming) choice. The transaction in that test case doesn't do anything more than exercising a consuming choice and there are many logic bugs that can cause it to fail, not just recording the key mapping later than at the beginning of the transaction.

I considered adding a comment about that specific case but again, in hindsight, I think it would not be clear to the reader why we chose to call out that one and not others.

Reply:
Thank you for clarification. I agree with you now. 👍

Reply:
I made "consuming" explicit though.

Reply:
I've seen this change - thanks!